src/App.tsx

```diff
@@ -1,26 +1,205 @@
+    const completedTodosId: number[] = todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+
+    Promise.allSettled(
+      completedTodosId.map(async id => {
+        await fetchDeleteTodo(id);
+      }),
+    );
+  }
+
+  async function fetchUpdateTodoCompleted(
+    todoId: number,
+    isCompleted: boolean,
+  ) {
+    try {
+      setUpdatedTodosId(current => [...current, todoId]);
+      await updateTodoCompleted(todoId, isCompleted);
+      setTodos(currentTodos =>
+        currentTodos.map(todo => {
+          if (todo.id === todoId) {
+            return { ...todo, completed: isCompleted };
+          }
+
+          return todo;
+        }),
+      );
+    } catch {
+      setErrorMessage(Errors.updateError);
+    } finally {
+      setUpdatedTodosId(current => current.filter(id => id !== todoId));
+    }
+  }
+
+  async function UpdateAllTodosCompleted() {
+    if (!countOfNotCompletedTodos) {
+      Promise.allSettled(
+        todos.map(async todo => {
+          await fetchUpdateTodoCompleted(todo.id, false);
+        }),
+      );
+    } else {
+      const notCompletedTodosId: number[] = todos
+        .filter(todo => !todo.completed)
+        .map(todo => todo.id);
+
+      Promise.allSettled(
+        notCompletedTodosId.map(async todoId => {
+          await fetchUpdateTodoCompleted(todoId, true);
+        }),
+      );
+    }
+  }
+
+  async function fetchUpdateTodoTitle(todoId: number, newTitle: string) {
+    const trimedNewTitle = newTitle.trim();
+
+    try {
+      setUpdatedTodosId(current => [...current, todoId]);
+      await updateTodoTitle(todoId, trimedNewTitle);
+      setTodos(currentTodos =>
+        currentTodos.map(todo => {
+          if (todo.id === todoId) {
+            return { ...todo, title: trimedNewTitle };
+          }
+
+          return todo;
+        }),
+      );
+
+      return true;
+    } catch {
+      setErrorMessage(Errors.updateError);
+
+      return false;
+    } finally {
+      setUpdatedTodosId(current => current.filter(id => id !== todoId));
+    }
+  }
+
+  useEffect(() => {
+    fetchTodosFromApi();
+  }, []);
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoHeader
+          countOfTodos={todos.length}
+          countOfCompletedTodos={countOfCompletedTodos}
+          isInputDisabled={!!tempTodo}
+          fetchAddTodo={fetchAddTodo}
+          setErrorMessage={setErrorMessage}
+          UpdateAllTodosCompleted={UpdateAllTodosCompleted}
```

Comment:
When passing props from a parent component to a child component, it's important to follow clear naming conventions that convey the purpose of the props, especially for handler functions.

Use handle prefix for event handlers in the parent and on prefix in the child.